src/lib.rs

```diff
@@ -586,6 +595,16 @@ pub struct Arrow {
 use self::ArrowShape::*;
 
 impl Arrow {
+    fn is_none(&self) -> bool {
+        self.arrows == &[NoArrow]
+    }
+
+    fn is_default_end_for_kind(&self, kind: Kind) -> bool {
```

Comment:
If we create a 'default' arrow type like I suggest above, then this can go away.